src/reachy2_sdk/reachy_sdk.py

```diff
@@ -476,27 +507,37 @@ def audit(self) -> Dict[str, Dict[str, str]]:
         return audit_dict
 
     def turn_on(self) -> bool:
-        """Activate all motors of the robot's parts.
+        """Activate all motors of the robot's parts if all of them are not already turned on.
 
         Returns:
             `True` if successful, `False` otherwise.
         """
-        speed_limit_high = 25
-
         if not self._grpc_connected or not self.info:
             self._logger.warning("Cannot turn on Reachy, not connected.")
             return False
-        for part in self.info._enabled_parts.values():
-            part.set_speed_limits(1)
-        time.sleep(0.05)
-        for part in self.info._enabled_parts.values():
-            part._turn_on()
-        if self._mobile_base is not None:
-            self._mobile_base._turn_on()
-        time.sleep(0.05)
-        for part in self.info._enabled_parts.values():
-            part.set_speed_limits(speed_limit_high)
-        time.sleep(0.4)
+
+        speed_limit_high = 25
+        success = self.is_on() and (
+            (not self._l_arm or not self._l_arm.gripper or self._l_arm.gripper.is_on())
+            and (not self._r_arm or not self._r_arm.gripper or self._r_arm.gripper.is_on())
+        )
+
+        while not success:
```

Comment:
Factorize `success = ...`
If can go to a function f() -> bool, and then the condition here would be 
`while not f():`

Reply:
Also a timeout, or a max number of attempt would be safer